keyboards/bpiphany/pegasushoof/2015/matrix.c

```diff
@@ -0,0 +1,168 @@
+/*
+  Copyright 2014 Ralf Schmitt <ralf@bunkertor.net>
+  Copyright 2016 Daniel Svensson <dsvensson@gmail.com>
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <avr/io.h>
+#include <util/delay.h>
+#include "wait.h"
+#include "print.h"
+#include "debug.h"
+#include "util.h"
+#include "matrix.h"
+
+static uint8_t debouncing = DEBOUNCE;
+static matrix_row_t matrix[MATRIX_ROWS];
+static matrix_row_t matrix_debouncing[MATRIX_ROWS];
+
+static matrix_row_t read_cols(void);
+static void select_row(uint8_t col);
+
+inline uint8_t matrix_rows(void)
+{
+  return MATRIX_ROWS;
+}
+
+inline uint8_t matrix_cols(void)
+{
+  return MATRIX_COLS;
+}
+
+void matrix_init(void)
+{
+  /* Column output pins */
+  DDRD  |=  0b01111011;
+  /* Row input pins */
+  DDRC  &= ~0b10000000;
+  DDRB  &= ~0b01111111;
+  PORTC |=  0b10000000;
+  PORTB |=  0b01111111;
+
+  for (uint8_t i=0; i < MATRIX_ROWS; i++)  {
+    matrix[i] = 0;
+    matrix_debouncing[i] = 0;
+  }
+
+  matrix_init_quantum();
+}
+
+uint8_t matrix_scan(void)
```

Comment:
Could this be updated to use the newer deboucing API?

Eg: 

https://github.com/qmk/qmk_firmware/blob/master/quantum/matrix.c#L310-L330

Reply:
OK, can you check this code? I have no idea what it does or if it's the right thing to do, it just seems to work.

Reply:
well, for now, then, it should be fine to leave, as is.